metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -893,6 +897,16 @@ def build_sink_node(
                 parent_node=pre_result_node or parent_node, include_specs=output_selection_specs
             )
 
+        alias_specs = tuple(
+            SpecToAlias(MetricSpec(metric.element_name), MetricSpec(metric.alias))
```

Comment:
For the spec with the alias, you should be able to just leave it as `metric` instead of creating a new metric spec because it will still be rendered with the alias.